### PR TITLE
Generalize symbolic helper code so it can be used more easily with other tests

### PR DESCRIPTION
**Pull request by majosm**
Mostly minor changes here; the only real noteworthy change is the addition of ~~the `MathMapper` class, which is intended to be a numpy-like object~~ the math module and its `__getattr__` function that dispatches math function calls to either `np`, `np.actx`, or `pymbl.var(...)` depending on the types of the arguments.

I'm not sure if having the math mapper be a module instance is the best way to implement this, so I'm open to changing it if there's another way that's cleaner. I mainly just wanted to be able to `import mirgecom.math as mm` and then `mm.sin(...)` instead of having to `from mirgecom.math import sin`, etc. for each function used. Maybe this isn't such a big deal? I dunno. 🤷‍♂️ 

**Questions for the review**:
- [ ] Is the scope and purpose of the PR clear?
  - [ ] The PR should have a description.
  - [ ] The PR should have a guide if needed (e.g., an ordering).
- [ ] Is every top-level method and class documented? Are things that should be documented actually so?
- [ ] Is the interface understandable? (I.e. can someone figure out what stuff does?) Is it well-defined?
- [ ] Does the implementation do what the docstring claims?
- [ ] Is everything that is implemented covered by tests?
- [ ] Do you see any immediate risks or performance disadvantages with the design? Example: what do interface normals attach to?

### CODE REVIEW

File: mirgecom/math.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+from pymbolic.primitives import Expression
+from arraycontext import get_container_context_recursively
+
+
+class _SymbolicMathContainer:
+    def __getattr__(self, name):
+        return pmbl.var(name)
+
+
+class MathMapper:
```
Comment (inducer) [inline mirgecom/math.py:41]:
```suggestion
class MathFunctionDispatcher:
```

File: mirgecom/math.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    """Dispatcher for math operations on symbolic/numeric input."""
```
Comment (inducer) [inline mirgecom/math.py:42]: Needs more:

* What's it for?
* How is it used?

File: mirgecom/math.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        if any(isinstance(arg, Expression) for arg in args):
+            return _SymbolicMathContainer()
+        else:
+            actx = get_container_context_recursively(make_obj_array(args))
+            if actx is not None:
+                return actx.np
+            else:
+                return np
+
+    def __getattr__(self, name):
```
Comment (inducer) [inline mirgecom/math.py:55]: Maybe use module-level `__getattr__`.

File: mirgecom/math.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
+"""Utilities and functions for math.
+
+.. autofunction:: __getattr__
```
Comment (inducer) [inline mirgecom/math.py:3]: Add a usage example.

**Review by inducer — APPROVED** [review]
Thanks! LGTM with an example added.

File: mirgecom/math.py (hunk 4 above)
Comment (majosm) [reply mirgecom/math.py:27]: Added.

**Review by inducer — APPROVED** [review]
:tada: LGTM, thanks!

**Pull request merged**